install/kubernetes/cilium/templates/cilium-operator/clusterrole.yaml

```diff
@@ -163,6 +163,9 @@ rules:
   verbs:
   # To synchronize garbage collection of such resources
   - update
+  {{- if (or (eq .Values.ciliumIdentityManagementMode "operator") (eq .Values.ciliumIdentityManagementMode "both")) }}
+  - create
```

Comment:
Kept `create` permission on cilium-agent for now out of concern that removing it could break cilium-agent when transitioning from "agent" -> "operator" mode.